pkg/cmd/release-payload-controller/payload_accepted_controller.go

```diff
@@ -133,9 +157,58 @@ func (c *PayloadAcceptedController) sync(ctx context.Context, key string) error
 		return err
 	}
 
+	acceptedCondition := computeReleasePayloadAcceptedCondition(originalReleasePayload)
+
 	releasePayload := originalReleasePayload.DeepCopy()
+	conditions.SetCondition(&releasePayload.Status.Conditions, acceptedCondition)
+	releasepayloadhelpers.CanonicalizeReleasePayloadStatus(releasePayload)
+
+	if reflect.DeepEqual(originalReleasePayload, releasePayload) {
+		return nil
+	}
 
 	klog.V(4).Infof("Syncing Payload Accepted for ReleasePayload: %s/%s", releasePayload.Namespace, releasePayload.Name)
+	_, err = c.releasePayloadClient.ReleasePayloads(releasePayload.Namespace).UpdateStatus(ctx, releasePayload, metav1.UpdateOptions{})
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
+
+func computeReleasePayloadAcceptedCondition(payload *v1alpha1.ReleasePayload) metav1.Condition {
+	acceptedCondition := metav1.Condition{
+		Type:   v1alpha1.ConditionPayloadAccepted,
+		Status: metav1.ConditionUnknown,
+		Reason: ReleasePayloadAcceptedReason,
+	}
+
+	// Check for "Accepted" PayloadOverride
+	switch payload.Spec.PayloadOverride.Override {
+	case v1alpha1.ReleasePayloadOverrideAccepted:
+		acceptedCondition.Status = metav1.ConditionTrue
+		acceptedCondition.Message = payload.Spec.PayloadOverride.Reason
+		acceptedCondition.Reason = ReleasePayloadManuallyAcceptedReason
+		return acceptedCondition
+	case v1alpha1.ReleasePayloadOverrideRejected:
+		// If a payload has already been rejected, then we do not want to process it any further...
+		acceptedCondition.Status = metav1.ConditionFalse
```

Comment:
you forgot to set reason and message

Reply:
Reason is set above L186.  Is Message required if I'm just blanking the field?

Reply:
Done!